### PR TITLE
Make the Skia expectation parser more resilient and update to "master_str"

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Piinks** [comment]
It looks like a test needs updating, and that for come reason Cirrus did not run the relevant tests. The failing test only became apparent from the Luci builds. I'll check the cirrus.yaml file

File: packages/flutter_goldens_client/lib/skia_client.dart [hunk 1]
```diff
@@ -438,20 +438,29 @@ class SkiaGoldClient {
       final Uri requestForExpectations = Uri.parse(
         'https://flutter-gold.skia.org/json/expectations/commit/HEAD'
       );
+      const String mainKey = 'master_str';
```
Comment (Piinks) [inline packages/flutter_goldens_client/lib/skia_client.dart:441]: Per the gold team, the current refactor they are doing will make `master` -> `master_str` (where we are now) -> `master`
Can we instead check to see if `master` is available, and if not use `master_str`? That would reduce churn and eliminate breakage for us in the interim.

**Comment by Piinks** [comment]
cc/ @LeandroLovisolo who is waiting on this to land to move forward with the refactoring

File: packages/flutter_goldens_client/lib/skia_client.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter_goldens_client/lib/skia_client.dart:441]: SGTM. Can we encourage them to avoid the term "master" entirely?

Comment (Piinks) [reply packages/flutter_goldens_client/lib/skia_client.dart:441]: I think it is based off our own branch naming convention. @LeandroLovisolo is that correct?

**Comment by Hixie** [comment]
@Piinks PTAL. I made the change you asked for. I also added a test, and silenced the existing tests (tests should not have output).

File: packages/flutter_goldens/test/json_templates.dart [hunk 2]
```diff
@@ -58,6 +58,29 @@ Map<String, List<String>> expectationsTemplate() {
   };
 }
 
+/// Same as [rawExpectationsTemplate] but with the temporary key.
+String rawExpectationsTemplateWithTemporaryKey() {
+  return '''
+    {
+      "md5": "a7489b00e03a1846e43500b7c14dd7b0",
+      "master": {
```
Comment (Piinks) [inline packages/flutter_goldens/test/json_templates.dart:66]:
```suggestion
      "master_str": {
```

**Review by Piinks — APPROVED** [review]
LGTM! Thanks!

File: packages/flutter_goldens/test/flutter_goldens_test.dart [hunk 3]
```diff
@@ -34,6 +34,17 @@ const List<int> _kFailPngBytes =
   120, 1, 99, 249, 207, 240, 255, 63, 0, 7, 18, 3, 2, 164, 147, 160, 197, 0,
   0, 0, 0, 73, 69, 78, 68, 174, 66, 96, 130];
 
+void testWithOutput(String name, Future<void> body(), String expectedOutput) async {
```
Comment (Piinks) [inline packages/flutter_goldens/test/flutter_goldens_test.dart:37]: To satisfy the analyzer:
```suggestion
Future<void> testWithOutput(String name, Future<void> body(), String expectedOutput) async {
```

**Pull request merged**